src/main/scala/com/casino/actors/WalletActor.scala

```diff
@@ -29,18 +40,22 @@ class WalletActor extends Actor with ActorLogging {
     case Deposit(id, amount) if state.contains(id) => {
       log.info(s"Deposit for user: $id with amount $amount")
       val newBalance = state(id) + amount
-      state += id -> newBalance
 
-      sender() ! Right(newBalance)
+      persist(Deposited(id, newBalance))(evt => {
+        updateState(evt)
+        sender() ! Right(newBalance)
```

Comment:
If this block of code is being executed asynchronously then usage of sender method is a mistake (since it changes).

Reply:
Hmm, sound reasonable. So it should be like this?
```scala
val requester = sender
persist(Deposited(id, newBalance))(evt => {
       updateState(evt)
        requester ! Right(newBalance)
}
```

Reply:
Yep.